### PR TITLE
[CH-402]Enable return null and complex type for get_json_object function

### PR DESCRIPTION
**Pull request by KevinyhZou**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Enable return null and complex type for get_json_object function. By adapt to 2 settings `function_json_value_return_type_allow_nullable` and  `function_json_value_return_type_allow_complex` as introduced in pr : https://github.com/ClickHouse/ClickHouse/pull/47494. The issue is described in https://github.com/Kyligence/ClickHouse/issues/402

### CODE REVIEW

**Comment by kyligence-git** [comment]
Can one of the admins verify this patch?

File: utils/local-engine/Parser/SerializedPlanParser.cpp [hunk 1]
```diff
@@ -1470,6 +1470,12 @@ const ActionsDAG::Node * SerializedPlanParser::parseFunctionWithDAG(
                 SerializedPlanParser::parseType(rel.scalar_function().output_type())->getName(),
                 function_node->result_name);
         }
+        if (function_name == "JSON_VALUE")
+        {
+            context->setSetting("function_json_value_return_type_allow_complex", true);
+            context->setSetting("function_json_value_return_type_allow_nullable", true);
```
Comment (liuneng1994) [inline utils/local-engine/Parser/SerializedPlanParser.cpp:1476]: 为什么要在这个位置修改配置。。不可以在最开始的时候配置好吗。在解析计划的时候修改context似乎不太合理

File: utils/local-engine/Common/QueryContext.h [hunk 2]
```diff
@@ -17,7 +17,7 @@ struct NativeAllocatorContext
 {
     std::shared_ptr<DB::CurrentThread::QueryScope> query_scope;
     std::shared_ptr<DB::ThreadStatus> thread_status;
-    DB::ContextPtr query_context;
+    DB::ContextMutablePtr query_context;
```
Comment (liuneng1994) [inline utils/local-engine/Common/QueryContext.h:20]: 不应该在这里修改context

Comment (KevinyhZou) [reply utils/local-engine/Common/QueryContext.h:20]: OK

**Comment by liuneng1994** [comment]
test this please

**Review by liuneng1994 — APPROVED** [review]
LGTM

**Pull request merged**